src/http/Message.cc

```diff
@@ -267,11 +267,16 @@ Http::Message::persistent() const
          */
         static SBuf close("close", 5);
         return !httpHeaderHasConnDir(&header, close);
-    } else {
+    }
+#if USE_HTTP_VIOLATIONS
+    else {
         /* for old versions of HTTP: persistent if has "keep-alive" */
         static SBuf keepAlive("keep-alive", 10);
         return httpHeaderHasConnDir(&header, keepAlive);
     }
+#endif
+    // RFC 7230 section 6.3: MUST NOT maintain a persistent connection with an HTTP/1.0 client
+    return false;
```

Comment:
Please structure these things such that Coverity etc. do not start complaining about unreachable code.

I expect that will mean inverting the violations condition to wrap the origin case checks, and make the "return false" be the thing wrapped / only exposed for strict builds.